### PR TITLE
Fix(eos_designs): Fix incorrect syntax in EVPN multicast PIM error messages

### LINKED ISSUE #3454: AnsibleMapping object has no attribute 'name' when enabling PIM with no RP
### Issue Summary

When configuring `l3_interfaces` under VRF with PIM enabled and no RP (`pim_rp_addresses`) configured in this VRF, ansible returns an error message which is a bit cryptic

```
fatal: [dc1-blm1a-mrv356 -> localhost]: FAILED! => \
{"changed": false, "msg": "'AnsibleMapping' object has no attribute 'name'"}
```

Already discussed with @jonxstill 

### Which component(s) of AVD impacted

eos_designs

### How do you run AVD ?

Ansible CLI (with virtual-env or native python)

### Steps to reproduce

```yaml
tenants:
  # Definition of tenants. Additional level of abstraction to VRFs
  - name: Tenant-A
    evpn_l3_multicast:
      evpn_underlay_l3_multicast_group_ipv4_pool: 232.232.0.0/20
      evpn_underlay_l3_multicast_group_ipv4_pool_offset: 1
    mac_vrf_vni_base: 10000
    vrfs:
      - name: MCAST
        vrf_id: 21
        vrf_vni: 21
        vtep_diagnostic:
          loopback: 21
          loopback_description: vtep_diagnostic
          loopback_ip_range: 10.254.0.0/24
        evpn_l3_multicast: 
          enabled: true
        svis:
          # List of svis truncated for comprehension
          - id: 219
            name: Flow_Multicast
            profile: GENERIC
            tags: [DC1]
            evpn_l2_multicast:
              enabled: false
            ip_address_virtual: 100.20.219.1/24
            ipv6_address_virtuals: ["fc00:100:20:219::1/64"]
        l3_interfaces:
          - nodes: [ dc1-blm1a-mrv356, dc1-blmba-mrv357 ]
            interfaces: [ Ethernet49/1.900, Ethernet49/1.900 ]
            encapsulation_dot1q_vlan: [ 900, 900 ]
            ip_addresses: [ 100.254.248.0/31, 100.254.248.4/31 ]
            descriptions: [ P2P_LINK_TO_rp-mrv380_Ethernet49/1, P2P_LINK_TO_rp-mrv380_Ethernet50/1 ]
            enabled: true
            pim:
              enabled: true 
          - nodes: [ dc1-blm1a-mrv356, dc1-blmba-mrv357 ]
            interfaces: [ Ethernet1.901, Ethernet1.901 ]
            encapsulation_dot1q_vlan: [ 901, 901 ]
            ip_addresses: [ 100.254.248.2/31, 100.254.248.6/31 ]
            descriptions: [ P2P_LINK_TO_row-hs307_Ethernet2, P2P_LINK_TO_row-hs307_Ethernet3 ]
            enabled: true
            pim:
              enabled: true
```

### Relevant log output

```shell
# Log from eos_design in task "Generate device configuration in structured format"

An exception occurred during task execution. To see the full traceback, use -vvv. The error was: AttributeError: 'AnsibleMapping' object has no attribute 'name'
fatal: [dc1-blm1a-mrv356 -> localhost]: FAILED! => {"changed": false, "msg": "'AnsibleMapping' object has no attribute 'name'"}
```

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by jonxstill**
## Change Summary

Mistakenly used Ansible "." syntax for accessing dict keys - should be "[]" in error messages.

## Related Issue(s)

Fixes #3454 

## Component(s) name

`arista.avd.eos_designs`

## How to test
Molecule negative tests added for this.

### User Checklist

- N/A

### Repository Checklist

- [X] My code has been rebased from devel before I start
- [X] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [ ] My change requires a change to the documentation and documentation have been updated accordingly.
- [X] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/ethernet_interfaces.py [hunk 1]
```diff
@@ -123,13 +123,13 @@ def ethernet_interfaces(self) -> list | None:
                                 if not vrf.get("_evpn_l3_multicast_enabled"):
                                     raise AristaAvdError(
                                         f"'pim: enabled' set on l3_interface {interface_name} on {self.shared_utils.hostname} requires evpn_l3_multicast:"
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/ethernet_interfaces.py]: let's add single quotes around `interface_name` and `self.shared_utils.hostname` as well for consistency

**Review by ClausHolbechArista — APPROVED** [review]
As long as CI passes LGTM! Thanks!

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/ethernet_interfaces.py (hunk 1 above)
Comment (jonxstill) [reply ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/ethernet_interfaces.py]: Fixed, thanks!

**Review by gmuloc — APPROVED** [review]
:shipit: LGTM once CI finishes

**Pull request merged**